### ISSUE TITLE
checkpoint_best.pt is not saved when finetuing with new datasets

### ISSUE THREAD

**Issue by mjeensung** [pytorch/fairseq#1302]
I've trained a transformer NMT model with iwslt17 dataset. I loaded the pretrained model using "--restore-file" argument and then tried to fine-tune it with more noisy dataset, which is "MTNT".

While training, checkpoint{epoch} and checkpoint_last are saved well.
However, checkpoint_best not being saved.

I found out that it's because fairseq_train compared the best performance from pretrained and the performance with new validation set, which is noisy text in this case, was always worse.

I want to save 'checkpoint_best' based only on the performance from new validation set independently. Is there a way to init the best performance from pretrained checkpoint?

**Comment by myleott** [comment]
You can do either `--reset-meters` or `--reset-optimizer`: https://github.com/pytorch/fairseq/blob/master/fairseq/checkpoint_utils.py#L118-L124
